File: tests/java/ComplexConditionalTest/ComplexConditionalTest1.java
// The class has method that has at least one if-else IE where: #(Expression(IE)) >= 3 (example with only "and").
// -> Has complex conditional smell.
public class ComplexConditionalTest1 //Type originates from: ComplexConditionalTest1.h
{
	private void func(int p1, int p2, int p3, int p4) //Method definition originates from: ComplexConditionalTest1.cpp
	{
			if (p1 > p4 && p1 > p2 && p1 > p3)
			{
				p1 = 0;
			}
	}
}

File: tests/cpp/ComplexConditionalTest/ComplexConditionalTest1.cpp
// The class has method that has at least one if-else IE where: #(Expression(IE)) >= 3 (example with only "and").
// -> Has complex conditional smell.
#include "ComplexConditionalTest1.h"
void ComplexConditionalTest1 :: func(int p1, int p2, int p3, int p4) {
        if (p1 > p4 && p1 > p2 && p1 > p3) {
            p1 = 0;
        }
}